fvm/state/state_holder.go

```diff
@@ -33,7 +36,8 @@ func (s *StateHolder) SetActiveState(st *State) {
 
 // SetPayerIsServiceAccount sets if the payer is the service account
 func (s *StateHolder) SetPayerIsServiceAccount() {
-	s.payerIsServiceAccount = true
+	s.activeState.SetTotalMemoryLimit(math.MaxUint64)
```

Comment:
The fact that this is only on active state worries me a bit.

If this is not done on the base state it will cause issues when merging the child state (with high limits) into the parent state with low limits. Maybe these limits should already be set when creating the root state.

This will also be invalidated [here](https://github.com/onflow/flow-go/blob/5119cb5/fvm/transactionEnv.go#L172) if the limit is set in the state.

Reply:
good catch